content/en/docs/concepts/scheduling-eviction/pod-eviction.md

```diff
@@ -0,0 +1,397 @@
+on those pods to `Guaranteed`.
+
+The kubelet uses various parameters to make eviction decisions, like the following:
+
+  * Eviction signals
+  * Eviction thresholds
+  * Monitoring intervals
+  * Node conditions
+
+## Triggering eviction
+
+You can configure pod eviction to happen automatically, triggered by a controller. 
+
+You can also request pod eviction manually using the `kubectl drain` command
+or the [Eviction API](/docs/reference/generated/kubernetes-api/v1.21/#create-eviction-pod-v1-core). 
+
+## Eviction signals {#eviction-signals}
+
+Eviction signals are the current state of a particular resource at a specific
+point in time. Kubelet uses eviction signals to make eviction decisions by
+comparing the signals to eviction thresholds, which are the minimum amount of 
+the resource that should be available on the node. 
+
+Kubelet uses the following eviction signals:
+
+| Eviction Signal      | Description                                                                           |
+|----------------------|---------------------------------------------------------------------------------------|
+| `memory.available`   | `memory.available` := `node.status.capacity[memory]` - `node.stats.memory.workingSet` |
+| `nodefs.available`   | `nodefs.available` := `node.stats.fs.available`                                       |
+| `nodefs.inodesFree`  | `nodefs.inodesFree` := `node.stats.fs.inodesFree`                                     |
+| `imagefs.available`  | `imagefs.available` := `node.stats.runtime.imagefs.available`                         |
+| `imagefs.inodesFree` | `imagefs.inodesFree` := `node.stats.runtime.imagefs.inodesFree`                       |
+| `pid.available`      | `pid.available` := `node.stats.rlimit.maxpid` - `node.stats.rlimit.curproc`           |
+
+In this table, the `Description` column shows how kubelet gets the value of the
+signal. Each signal supports either a percentage or a literal value. Kubelet 
+calculates the percentage value relative to the total capacity associated with
+the signal. 
+
+The value for `memory.available` is derived from the cgroupfs instead of tools
+like `free -m`. This is important because `free -m` does not work in a
+container, and if users use the [node
+allocatable](/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable) feature, out of resource decisions
+are made local to the end user Pod part of the cgroup hierarchy as well as the
+root node. This [script](/examples/admin/resource/memory-available.sh)
+reproduces the same set of steps that the `kubelet` performs to calculate
+`memory.available`. The `kubelet` excludes inactive_file (i.e. # of bytes of
+file-backed memory on inactive LRU list) from its calculation as it assumes that
+memory is reclaimable under pressure.
+
+The kubelet supports the following filesystems:
+
+1. `nodefs`: Volumes, daemon logs, etc.
+1. `imagefs`: Container runtimes use `imagefs` to store container images and
```

Comment:
What are nodefs and imagefs? (For example, would I find these names in `/proc/mounts`?)

(I think these are symbolic names used within the kubelet and its configuration, and not used elsewhere). We should clarify what these names mean.

Reply:
`nodefs` and `imagefs` do seem like the symbolic names for the node's root filesystem (nodefs) and the runtime filesystem (imagefs), which in Docker is apparently "thin pool". 

I reworded these points to: 

1. `nodefs`: The node's root filesystem, used for local disk volumes, emptyDir, daemon logs, and more.
1. `imagefs`: An optional filesystem that container runtimes use to store images and writable layers.

Reply:
Further nit: when Kubernetes people say “node's root filesystem” they mean “filesystem containing `/var/lib/kubelet`” which may not be the filesystem that contains `/`.
I'm not sure if there's a good name for it though. “Node's root filesystem” is nice and short, just sometimes wrong.

Reply:
@dashpole or @bobbypage can you confirm?

Reply:
The above are all correct.

Reply:
changed to 

1. `nodefs`: The node's root filesystem, used for local disk volumes, emptyDir,
   log storage, and more. For example, `nodefs` can provide `/` or `/var/lib/kubelet`. 

Does that cover potential confusion?

Reply:
Strictly speaking _nodefs_ refers to the filesystem containing the kubelet's main directory (that is, `/var/lib/kubelet`). If we can avoid writing `/` or mentioning “root” that avoids a possible point of confusion.

Reply:
Updated to 1. `nodefs`: The node's main filesystem, used for local disk volumes, emptyDir,
   log storage, and more. For example, `nodefs` contains `/var/lib/kubelet/`.